### PR TITLE
feat: [IDP-187]: API support for Account to NS and Git Integration

### PR DESCRIPTION
**Pull request by devesh-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-ALL: `trigger tiAll`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/44126)

### CODE REVIEW

**Review by deepharness — DISMISSED** [review]
> Reviewed 5 of 14 files at r1, 6 of 9 files at r2, 4 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44126)* status: 15 file(s) to review. 1 of 2 approvals obtained.

File: idp-service/service/src/main/java/io/harness/idp/namespace/beans/dto/Namespace.java [hunk 1]
```diff
@@ -1,26 +0,0 @@
...
-import lombok.Builder;
-import lombok.Getter;
-import org.codehaus.jackson.annotate.JsonIgnoreProperties;
-
-@Builder
-@Getter
-@JsonIgnoreProperties(ignoreUnknown = true)
-@JsonInclude(JsonInclude.Include.NON_NULL)
-@OwnedBy(HarnessTeam.IDP)
-public class Namespace {
```
Comment (vigneswara-propelo) [inline idp-service/service/src/main/java/io/harness/idp/namespace/beans/dto/Namespace.java:23]: I think you might have to take the latest code and resolve conflicts as there are few places (like EnvironmentSecretServiceImpl & PodHealthCheck) that still use this class

Comment (devesh-harness) [reply idp-service/service/src/main/java/io/harness/idp/namespace/beans/dto/Namespace.java:23]: Done, yes this was due to latest change merges.

**Review by vigneswara-propelo — APPROVED** [review]
> Reviewed 5 of 14 files at r1, 6 of 9 files at r2, 1 of 4 files at r3, 5 of 5 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44126)* status: 5 file(s) to review. 1 of 2 approvals obtained.

**Review by deepharness — APPROVED** [review]
> Reviewed 5 of 5 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44126)* status: :shipit: complete! ✓

**Pull request merged**